flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/SearchOperatorGen.scala

```diff
@@ -96,7 +96,9 @@ object SearchOperatorGen {
            |boolean $nullTerm = true;
            |if (!${needle.nullTerm}) {
            |  $resultTerm = $negation$setTerm.contains(${needle.resultTerm});
-           |  $nullTerm = !$resultTerm && $setTerm.containsNull();
+           |  $nullTerm = false;
+           |} else {
+           |  $resultTerm = $setTerm.containsNull();
```

Comment:
You didn't set `$nullTerm`, so the result will always be `UNKNOWN`.

I actually expect something like below to make the code clearer.

```scala
val isNullCode = sarg.nullAs match {
  case RexUnknownAs.TRUE =>
    s"""
       |$resultTerm = true;
       |$nullTerm = false;
       |""".stripMargin
  case RexUnknownAs.FALSE =>
    s"""
       |$resultTerm = false;
       |$nullTerm = false;
       |""".stripMargin
  case RexUnknownAs.UNKNOWN =>
    s"""
       |$resultTerm = false;
       |$nullTerm = true;
       |""".stripMargin
}

// ...

|} else {
|  $isNullCode
```

Also add code comments on what `sarg.nullAs` means. It would be better to add the links you mentioned above in the code comments.

Reply:
thanks for bearing with me

I updated the code in a way similar to what you've mentioned above.